server/routes/api/index.js

```diff
@@ -0,0 +1,38 @@
+const apiRouter = require('express').Router();
+const { Octokit } = require('@octokit/rest');
+const debug = require('../../debug');
+
+const octokit = Octokit({
+    baseUrl: process.env.GITHUB_API_URL
+});
+
+apiRouter.get('/hell', (req, res) => res.send('123'));
+
+apiRouter.get('/pulls', async (req, res, next) => {
+    const options = octokit.pulls.list.endpoint.merge({
+        owner: process.env.OWNER,
+        repo:  process.env.REPO
+    });
+
+    const pullRequests = [];
+
+    try {
+        for await (const page of octokit.paginate.iterator(options))
+            pullRequests.push(page['data'].filter(pr => pr['author_association'] === 'NONE'));
+    }
+    catch (err) {
+        debug.error(`Error happened when ${req.ip} requests ${req.path}`);
+        debug.error(err);
+
+        const error = {
+            message: 'An error has occured while fetching pull-requests from server.',
+            status:  500
+        };
+
+        return next(error);
```

Comment:
Well, when writing request handlers it's better to always set up them as terminal handlers and don't pass handled requests to preserve simplicity.

Reply:
I pass error to middleware error-handler. 
```js
e.use((err, req, res, next) => {
    res.status(err.status).json(err);
    next();
});
```
In your opinion, better to handle error in place?

Reply:
Yup, it's better to handle it explicitly and immediately instead of relying on some middleware that is configured somewhere else. 

It's still a good idea to have a global error handler, but it's better to do not leave all the error handling jobs to it.

To avoid code duplication, you can write a separate function that will log the error and from a request to the client, extract it to a separate module and use it both in your global error handling middleware and router.

Reply:
Also, as we're talking about global error handling: it's a good idea to handle the `uncaughtException` and `unhandledRejection` events that can be emitted by `process`.